### ISSUE TITLE
Allow pip extra args for playbooks

### ISSUE THREAD

**Issue by odyssey4me** [rcbops/rpc-openstack#428]
In https://review.openstack.org/229542 OSA allows extra args to be passed to the pip install task in the roles. This may be a pattern which rpc-openstack wishes to apply for its own roles.

**Comment by byronmccollum** [comment]
This is now needed. Seeing upgrades failing without this.

**Comment by byronmccollum** [comment]
```
failed: [dcb7e5c2-block1] => (item=python-novaclient) => {"cmd": "/usr/local/bin/pip install python-novaclient", "failed": true, "item": "python-novaclient"}
msg: stdout: Ignoring indexes: https://pypi.python.org/simple
Collecting python-novaclient
  Downloading http://172.29.239.235:8181/os-releases/11.2.4/python_novaclient-2.23.2-py2.py3-none-any.whl (290kB)
Requirement already up-to-date: oslo.i18n<1.6.0,>=1.5.0 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: argparse in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: oslo.serialization<1.5.0,>=1.4.0 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: requests!=2.4.0,>=2.2.0 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: pbr!=0.7,<1.0,>=0.6 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: oslo.utils<1.5.0,>=1.4.0 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: iso8601>=0.1.9 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: python-keystoneclient<1.4.0,>=1.2.0 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: Babel>=1.3 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: six>=1.9.0 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: PrettyTable<0.8,>=0.7 in /usr/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: simplejson>=2.2.0 in /usr/local/lib/python2.7/dist-packages (from python-novaclient)
Requirement already up-to-date: msgpack-python>=0.4.0 in /usr/local/lib/python2.7/dist-packages (from oslo.serialization<1.5.0,>=1.4.0->python-novaclient)
Requirement already up-to-date: pytz>=2013.6 in /usr/local/lib/python2.7/dist-packages (from oslo.serialization<1.5.0,>=1.4.0->python-novaclient)
Requirement already up-to-date: pip in /usr/local/lib/python2.7/dist-packages (from pbr!=0.7,<1.0,>=0.6->python-novaclient)
Collecting netaddr<=0.7.13,>=0.7.12 (from oslo.utils<1.5.0,>=1.4.0->python-novaclient)

:stderr:   Could not find a version that satisfies the requirement netaddr<=0.7.13,>=0.7.12 (from oslo.utils<1.5.0,>=1.4.0->python-novaclient) (from versions: 0.7.18)
No matching distribution found for netaddr<=0.7.13,>=0.7.12 (from oslo.utils<1.5.0,>=1.4.0->python-novaclient)
```

**Comment by d34dh0r53** [comment]
I believe this was fixed with #549
